prow/cmd/pipeline/controller.go

```diff
@@ -352,6 +378,61 @@ func (c *controller) pipelineID(pj prowjobv1.ProwJob) (string, string, error) {
 	return id, url, nil
 }
 
+func createProwJobIdentifier(pj *prowjobv1.ProwJob) string {
+	var additionalIdentifier string
+	if pj.Spec.Refs != nil {
+		var pulls []int
+		for _, pull := range pj.Spec.Refs.Pulls {
+			pulls = append(pulls, pull.Number)
+		}
+		sort.Ints(pulls)
+		additionalIdentifier = fmt.Sprintf("%s/%s@%s %v", pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.BaseRef, pulls)
+	}
+	return fmt.Sprintf("%s %s", pj.Spec.Job, additionalIdentifier)
+}
+
+func getFilteredProwJobs(id string, pjsToFilter []*prowjobv1.ProwJob) []*prowjobv1.ProwJob {
+	pjs := []*prowjobv1.ProwJob{}
+	for _, p := range pjsToFilter {
+		if runningState(p.Status.State) && id == createProwJobIdentifier(p) {
+			pjs = append(pjs, p)
+		}
+	}
+	sort.Slice(pjs, func(i, j int) bool {
+		return pjs[i].Status.StartTime.Before(&pjs[j].Status.StartTime)
+	})
+	return pjs
+}
+
+func abortDuplicatedProwJobs(c reconciler, pj *prowjobv1.ProwJob) (*prowjobv1.ProwJob, error) {
+	if !runningState(pj.Status.State) || pj.Spec.Agent != prowjobv1.TektonAgent {
```

Comment:
what if we are still pending or something? we should still abort IMO.

Reply:
this function means any running so pending and triggered are counted as well